### ISSUE TITLE
Trying to mutate a frozen dict value in _merge_dart_context() in internal.bzl

### ISSUE THREAD

**Issue by timoffex** [cbracken/rules_dart#60]
I was trying again to use Bazel for a personal project using AngularDart and encountered the following error:

```
	File "/private/var/tmp/_bazel_timoffex/9da7457002ac89f9b1045d3059718093/external/io_bazel_rules_dart/dart/build_rules/internal.bzl", line 149, in _merge_dart_context
		transitive_deps.update(<1 more arguments>)
trying to mutate a frozen dict value
```

Based on https://docs.bazel.build/versions/0.24.0/skylark/language.html#mutability, I get the feeling that this shouldn't be updating the dictionary. The error goes away if I merge my `dart_library` dependencies into a single rule:

```
# original; causes error above
dart_library(
    name = "coreweb",
    srcs = glob(["lib/*"]),
    visibility = ["//visibility:public"],
    deps = [":_internal"],
)

dart_library(
    name = "_internal",
    srcs = glob(["lib/src/*"]),
    visibility = ["//visibility:private"],
)

# merged approach; works
dart_library(
    name = "coreweb"
    srcs = glob(["lib/**"]),
    visibility = ["//visibility:public"],
)
```

**Comment by timoffex** [comment]
In case it is helpful:

```
> bazel --version
bazel 3.3.1-homebrew
```

**Comment by cbracken** [comment]
Thanks. Fixed in #79.
